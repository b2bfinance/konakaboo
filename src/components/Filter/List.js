import Grid from '@material-ui/core/Grid';
import ArrowDropDown from '@material-ui/icons/ArrowDropDown';
import Cancel from '@material-ui/icons/Cancel';
import React, { useState } from 'react';
import styled from 'styled-components';
import {
  resetFilters,
  resetGroupFilters,
  setFilters
} from '../../actions/filter';
import { generateChipLabel } from '../../utils/filter';
import Chip, { BaseChip, ChipHolder } from './Chip';
import Filter from './Filter';

export const Wrapper = styled(Grid)`
  justify-content: flex-end;
  margin-bottom: 32px;
`;

export default ({ filters, dispatch }) => {
  const [activeGroup, setActiveGroup] = useState();

  const handleDeleteChosen = (group, key) => {
    if (filters.chosen[group].length) {
      if (activeGroup === key) {
        setActiveGroup();
      }

      dispatch(resetGroupFilters(group));
      return;
    }

    setActiveGroup(key);
  };

  return (
    <Wrapper container>
      {filters.available.map((filter, i) => (
        <ChipHolder key={filter.key}>
          <Chip
            selection={filters.chosen[i]}
            label={generateChipLabel(
              filter.title,
              filter.multiChoice,
              filters.chosen[i],
              filter.choices
            )}
            deleteIcon={
              filters.chosen[i].length ? <Cancel /> : <ArrowDropDown />
            }
            onClick={() => setActiveGroup(filter.key)}
            onDelete={() => handleDeleteChosen(i, filter.key)}
            onBlur={() => setActiveGroup()}
          />
          <Filter
            visible={activeGroup === filter.key}
            group={i}
            multi={filter.multiChoice}
            title={filter.title}
            choices={filter.choices}
            chosen={filters.chosen[i]}
            handleSetChosenForGroup={(group, chosen) =>
              dispatch(setFilters(group, chosen))
            }
            handleClose={() => setActiveGroup()}
          />
        </ChipHolder>
      ))}
      <BaseChip
        label="Reset Filters"
        onClick={() => dispatch(resetFilters())}
      />
    </Wrapper>
  );
};
