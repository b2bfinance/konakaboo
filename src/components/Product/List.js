import React from 'react';
import Product from './Product';
import Error from './Error';
import Loading, { Mask } from './Loading';
import Empty from './Empty';

export default ({ error, hasFetched, isFetching, cta, products }) => {
  if (error) {
    return (
      <Error message="We had problems retrieving products for you, retry or come back later." />
    );
  }

  if (!hasFetched && !products) {
    return <Mask />;
  }

  if (products.length === 0) {
    return <Empty />;
  }

  return (
    <Loading loading={isFetching}>
      {products.map((product, i) => (
        <Product key={i} cta={cta} {...product} />
      ))}
    </Loading>
  );
};
