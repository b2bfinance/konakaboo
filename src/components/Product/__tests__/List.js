import React from 'react';
import { shallow, render } from 'enzyme';
import List from '../List';

describe('List', () => {
  test('renders correctly when fetching has not started', () => {
    expect(render(<List hasFetched={false} />)).toMatchSnapshot();
  });

  test('renders correctly with an error', () => {
    expect(render(<List hasFetched={true} error={true} />)).toMatchSnapshot();
  });

  test('renders correctly with no products', () => {
    expect(render(<List hasFetched={true} products={[]} />)).toMatchSnapshot();
  });

  test('renders correctly with products', () => {
    expect(
      shallow(
        <List hasFetched={true} products={stubData.products.response.data} />
      )
    ).toMatchSnapshot();
  });
});
