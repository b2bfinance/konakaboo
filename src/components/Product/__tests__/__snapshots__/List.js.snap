// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly when fetching has not started 1`] = `
Array [
  .c0 {
  height: 35rem;
  margin-bottom: 1.5rem;
}

@media (min-width:62em) {
  .c0 {
    height: 9rem;
  }
}

<div
    class="c0"
  />,
  .c0 {
  height: 35rem;
  margin-bottom: 1.5rem;
}

@media (min-width:62em) {
  .c0 {
    height: 9rem;
  }
}

<div
    class="c0"
  />,
  .c0 {
  height: 35rem;
  margin-bottom: 1.5rem;
}

@media (min-width:62em) {
  .c0 {
    height: 9rem;
  }
}

<div
    class="c0"
  />,
]
`;

exports[`List renders correctly with an error 1`] = `
.c0 {
  background-color: #ff9800;
  color: #fff;
  padding: 1rem;
}

<div
  class="c0"
>
  We had problems retrieving products for you, retry or come back later.
</div>
`;

exports[`List renders correctly with no products 1`] = `
.c0 {
  margin: 2rem auto;
  padding: 2rem;
}

<div
  class="c0"
>
  <p>
    There are no product matching your requirements.
  </p>
</div>
`;

exports[`List renders correctly with products 1`] = `
<_default>
  <Product
    brand="Landbay"
    columns={
      Array [
        Object {
          "label": "Max LTV",
          "value": "85%",
        },
        Object {
          "label": "Initial Rate",
          "value": "1.49%",
        },
        Object {
          "label": "Subsequent Rate",
          "value": "2.75%",
        },
        Object {
          "label": "Fees",
          "value": "£1,495",
        },
      ]
    }
    description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean bibendum facilisis condimentum. Donec laoreet porttitor condimentum. Nulla luctus efficitur magna, eget rhoncus metus placerat sed."
    detailed={
      Object {
        "Highlight Points": Array [
          Object {
            "label": "Max LTV",
            "value": "85%",
          },
          Object {
            "label": "Initial Rate",
            "value": "1.49%",
          },
          Object {
            "label": "Subsequent Rate",
            "value": "2.75%",
          },
          Object {
            "label": "Fees",
            "value": "£1,495",
          },
        ],
        "Point List": Array [
          Object {
            "label": "",
            "value": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
          },
        ],
        "Technical Points": Array [
          Object {
            "label": "Max LTV",
            "value": "85%",
          },
          Object {
            "label": "Initial Rate",
            "value": "1.49%",
          },
          Object {
            "label": "Subsequent Rate",
            "value": "2.75%",
          },
          Object {
            "label": "Fees",
            "value": "£1,495",
          },
        ],
      }
    }
    disclaimer="Application for this product will impact your credit score."
    feature_point="Awarded best broker 2019"
    highlighted={true}
    key="0"
    labels={
      Array [
        "broker only",
        "consumer favourite",
      ]
    }
    links={
      Object {
        "apply": "https://b2bfinance.com/enquire/buy-to-let",
        "logo": "https://b2bfinance.com/images/partners/Landbay.png",
      }
    }
    meta={
      Object {
        "confirm": Object {
          "description": "B2Bfinance.com are registered intermediaries who will quote, package and submit your lending request to Natwest. There is no charge for this service as we are paid a small commission by Natwest to cover our time.",
          "heading": "Application via Brokers only",
        },
      }
    }
    title="Landbay - Buy To Let Wizards"
  />
</_default>
`;
